File: src/main/java/com/github/kindrat/cassandra/client/ui/window/editor/main/filter/FilterGrid.java
package com.github.kindrat.cassandra.client.ui.window.editor.main.filter;

import com.datastax.driver.core.TableMetadata;
import com.github.kindrat.cassandra.client.ui.DataObject;
import com.github.kindrat.cassandra.client.ui.eventhandler.FilterBtnHandler;
import com.github.kindrat.cassandra.client.ui.window.editor.main.table.DataTableView;
import javafx.collections.ObservableList;
import javafx.geometry.HPos;
import javafx.scene.control.Button;
import javafx.scene.layout.ColumnConstraints;
import javafx.scene.layout.GridPane;
import javafx.scene.layout.Priority;
import javafx.scene.layout.RowConstraints;

public class FilterGrid extends GridPane {
    private final Button filterButton;
    private final FilterTextField textField;
    private final DataTableView dataTableView;

    public FilterGrid(DataTableView dataTableView, FilterTextField textField) {
        this.dataTableView = dataTableView;
        this.filterButton = new Button("Filter");
        this.filterButton.setMnemonicParsing(false);
        this.textField = textField;

        ColumnConstraints column1 = new ColumnConstraints();
        column1.setPrefWidth(100);
        column1.setMinWidth(10);
        column1.setHalignment(HPos.CENTER);
        column1.setHgrow(Priority.SOMETIMES);

        ColumnConstraints column2 = new ColumnConstraints();
        column2.setPrefWidth(100);
        column2.setMinWidth(100);
        column2.setMaxWidth(100);
        column2.setHalignment(HPos.CENTER);
        column2.setHgrow(Priority.SOMETIMES);

        getColumnConstraints().clear();
        getColumnConstraints().addAll(column1, column2);

        RowConstraints row = new RowConstraints(10, 30, 30);
        row.setVgrow(Priority.SOMETIMES);

        getRowConstraints().clear();
        getRowConstraints().add(row);

        add(textField, 0, 0);
        add(filterButton, 1, 0);
    }

    public void setTableMetadata(TableMetadata tableMetadata) {
        textField.setTableMetadata(tableMetadata);
    }

    public void suggestCompletion() {
        if (textField.isFocused()) {
            textField.suggestCompletion();
        }
    }

    public void onDataUpdated(ObservableList<DataObject> data) {
        filterButton.setOnAction(new FilterBtnHandler(textField, dataTableView, data));
        textField.setOnAction(new FilterBtnHandler(textField, dataTableView, data));
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/window/editor/main/filter/FilterGrid.kt
package com.github.kindrat.cassandra.client.ui.window.editor.main.filter

import com.datastax.oss.driver.api.core.metadata.schema.TableMetadata
import com.github.kindrat.cassandra.client.ui.DataObject
import com.github.kindrat.cassandra.client.ui.eventhandler.FilterBtnHandler
import com.github.kindrat.cassandra.client.ui.window.editor.main.table.DataTableView
import javafx.collections.ObservableList
import javafx.geometry.HPos
import javafx.scene.control.Button
import javafx.scene.layout.ColumnConstraints
import javafx.scene.layout.GridPane
import javafx.scene.layout.Priority
import javafx.scene.layout.RowConstraints

class FilterGrid(private val dataTableView: DataTableView, private val textField: FilterTextField) : GridPane() {
    private val filterButton: Button = Button("Filter")

    fun setTableMetadata(tableMetadata: TableMetadata) {
        textField.setTableMetadata(tableMetadata)
    }

    fun suggestCompletion() {
        if (textField.isFocused) {
            textField.suggestCompletion()
        }
    }

    fun onDataUpdated(data: ObservableList<DataObject>) {
        filterButton.onAction = FilterBtnHandler(textField, dataTableView, data)
        textField.onAction = FilterBtnHandler(textField, dataTableView, data)
    }

    init {
        filterButton.isMnemonicParsing = false
        val column1 = ColumnConstraints()
        column1.prefWidth = 100.0
        column1.minWidth = 10.0
        column1.halignment = HPos.CENTER
        column1.hgrow = Priority.SOMETIMES
        val column2 = ColumnConstraints()
        column2.prefWidth = 100.0
        column2.minWidth = 100.0
        column2.maxWidth = 100.0
        column2.halignment = HPos.CENTER
        column2.hgrow = Priority.SOMETIMES
        columnConstraints.clear()
        columnConstraints.addAll(column1, column2)
        val row = RowConstraints(10.0, 30.0, 30.0)
        row.vgrow = Priority.SOMETIMES
        rowConstraints.clear()
        rowConstraints.add(row)
        add(textField, 0, 0)
        add(filterButton, 1, 0)
    }
}